Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -0,0 +1,303 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Video;
+using Debug = UnityEngine.Debug;
+
+// This component uses code from the following sources:
+// UnityYoutubePlayer, courtesy iBicha (SPDX-License-Identifier: Unlicense) https://github.com/iBicha/UnityYoutubePlayer
+// USharpVideo, Copyright (c) 2020 Merlin, (SPDX-License-Identifier: MIT) https://github.com/MerlinVR/USharpVideo/
+
+namespace VRCAudioLink
+{
+    /// <summary> Downloads and plays videos via a VideoPlayer component </summary>
+    [RequireComponent(typeof(VideoPlayer))]
+    public class YtdlpPlayer : MonoBehaviour
+    {
+        /// <summary> Ytdlp url (e.g. https://www.youtube.com/watch?v=SFTcZ1GXOCQ) </summary>
+        public string ytdlpURL = "https://www.youtube.com/watch?v=SFTcZ1GXOCQ";
+
+        /// <summary> VideoPlayer component associated with the current YtdlpPlayer instance </summary>
+        public VideoPlayer VideoPlayer { get; private set; }
+
+        /// <summary> Initialize and play from URL </summary>
+        void OnEnable()
+        {
+            VideoPlayer = GetComponent<VideoPlayer>();
+            UpdateURL();
+            if (VideoPlayer.length > 0)
+                VideoPlayer.Play();
+        }
+
+        /// <summary> Update URL and start playing </summary>
+        public void UpdateAndPlay()
+        {
+            UpdateURL();
+            if (VideoPlayer.length > 0)
+                VideoPlayer.Play();
+        }
+
+        /// <summary> Set time to zero, resolve, and set URL </summary>
+        public void UpdateURL()
+        {
+            try
```

Comment:
this try catch also seems pointless? can anything fail but the innermost Resolve?

Reply:
Removed, resolved in 017fd35fa743a75b5ee7b0914fd4a2328bfc13d4